paprika/simulate.py

```diff
@@ -0,0 +1,160 @@
+import numpy as np
+import subprocess as sp
+import logging as log
+import os as os
+import parmed as pmd
+from collections import OrderedDict
+
+
+def _write_input_file(filename, input_dict, title='Input.'):
+    """
+    Process a dictionary to generate a pmemd input file.
+    """
+
+    log.debug('Writing {}'.format(filename))
+    with open(filename, 'w') as f:
+        f.write("{}\n".format(title))
+        for namelist in ['cntrl','ewald']:
+            if input_dict[namelist] is not None:
+                f.write(" &{}\n".format(namelist))
+                for key,val in input_dict[namelist].items():
+                    if val is not None:
+                        f.write("  {:15s} {:s},\n".format(key+' =',str(val)))
+                f.write(" /\n")
+        if input_dict['cntrl']['nmropt'] == 1:          ### Need to check if this exists first!!
```

Comment:
Should there be an accompanying `else` to this?

Reply:
Probably not, if nmropt != 1, then we don't need to do anything.  just finish the &cntrl block and be done.

Reply:
but this whole code may eventually need more robust handling of unexpected user inputs.

Reply:
Okay... but what do you mean by first? It needs to be checked before GROUP?

Reply:
That line will fail if `input_dict['cntrl']['nmropt'] == None`.  I mean, this is APR, so I don't think we'll ever NOT be using restraints, but I should check that the key is not `None`.

Reply:
oh wait, no it won't.  it will fail if the key hasn't been set.  probably not an issue since the code will always set it to `None`.

Reply:
Seems fine, if clunky.